server/models/user.js

```diff
@@ -1,11 +1,40 @@
 const mongoose = require('mongoose');
-
-const Schema = mongoose.Schema;
+const { Schema } = mongoose;
 
 const userSchema = new Schema({
-  // To Do: define user using mongoose.js
+  first_name: { type: String, required: true },
+  last_name: String,
+  password: { type: Number, required: true },
+  // Do we need reset ability yet?
+  resetToke: Number,
+  resetTokenExpiry: Date,
+  // To do: add validation
+  email: { 
+    type: String, 
+    required: true,
+    unique: true,
+  },
+  primaryAddress: {
+    street: { type: String, required: true },
+    city: { type: String, required: true },
+    region: { type: String, required: true },
+    postalCode: { type: Number, required: true },
+    country: { type: String, required: true },
+  },
+  // To do: add validation
+  primaryPhone: Number,
+  profilePicURL: String,
+  bio: String,
+  favoriteCuisine: [{
+    type:String,
+    length: 5,
```

Comment:
is this a required length or just a maximum?

Reply:
This was supposed to be the min length of the array but after relooking this is not how you do it.  I don't think we need to worry about since we are doing validation outside of the schema. I had put a min length because if the user picks a ton of foods they don't really have a favorite any more.